### PR TITLE
 Update to make the button a metabox Gutenberg compatible 

### PR DESCRIPTION
**Pull request by typometre**
Having a metabox is better to make it compatible with Gutenberg editor and to make it easier to provide more option in future version.
1bf81dc

### CODE REVIEW

**Comment by jonathanbossenger** [comment]
@arthurattwell I'd appreciate your thoughts on this. While converting the buttons to being part of a meta box does make it immediately Gutenberg compatible, it does move away from your original requirement of having the buttons close to the publish controls.

**Comment by typometre** [comment]
On Classic editor, you can move the metabox where you want. Could be higher than the publish/update wordpress metabox. On Gutenberg you can move the metabox but only to arrange the order with other site specific metabox, you can't place it before Wordpress default ones. In order to provide future options in the panel (like select or write css option), the metabox panel seems to me the way to go.

**Comment by jonathanbossenger** [comment]
Unless the requirement is to add the controls in the same place as the publish or preview buttons, in a similar way to which Jetpack adds it's button in the block editor, which may provide for a better user experience.

http://tinyurl.com/y394pudl

**Comment by typometre** [comment]
That's a sidebar plugin. It's a way to open a new panel in sidebar. Could be useful if eventually there will be too many options and need space for them. But it actually isn't a way to add a button in itself, but a button to open a complete new sidebar panel where you would place the paged preview button. Doesn't seems to help as it would require more steps to go to the paged preview button. https://developer.wordpress.org/block-editor/tutorials/plugin-sidebar-0/
There is plugins that adds buttons at that top place which aren't sidebar plugin (as Elementor add an edit with elementor on the left side) but this kind of things seems undocumented on official gutenberg docs.

**Comment by jonathanbossenger** [comment]
@typometre the type of button is not the point I was trying to make, the initial requirement for the Paged Preview button was to have it appear as close to the publish buttons as possible. I was merely using the Jetpack button as a visual example of this in the block editor. As you've pointed out, other plugins have added their own buttons to the block editor top bar and if the requirement remains the same, it's worth investigating it, undocumented or not.

**Comment by arthurattwell** [comment]
Thanks, @typometre! Gutenberg compatibility will be great. Thanks, @jonathanbossenger, for engaging with this. We're planning a sprint on this in the coming weeks and will be able to make a call on this then, along with improvements we have planned in #5.

File: README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ There is a lot still to improve here. For now:
 
 - Paged.js only works properly in Chrome, so you should use Chrome for this, too. 
 - You must hard-code the link to your Paged Media CSS in the paged-wp.php file. By default, it loads the template styles from [paged.design](https://paged.design).
-- In WordPress, you must use the Classic editor. This plugin does not work with the Gutenberg editor.
+- This fork work with the Classic or Gutenberg editor.
```
Comment (jonathanbossenger) [inline README.md:13]: We can remove this.

File: paged-wp.php [hunk 2]
```diff
@@ -18,16 +18,27 @@
...
+ * Register Paged Preview meta box.
  */
+function wppaged_register_meta_boxes() {
+    add_meta_box( 'meta-box-paged', __( 'Paged Preview', 'textdomain' ), 'wppaged_my_display_callback', 'post', 'side', 'high',
+    array(
+        '__back_compat_meta_box' => false, 
+)
+			);
+}
+add_action( 'add_meta_boxes', 'wppaged_register_meta_boxes' );
```
Comment (jonathanbossenger) [inline paged-wp.php:33]: Please register the action before the callback function, that way it's easier to see which function is being called by which action hook.

File: paged-wp.php [hunk 3]
```diff
@@ -18,16 +18,27 @@
...
 define( 'PAGED_PLUGIN_URL', plugin_dir_url( __FILE__ ) );
-define( 'PAGED_VERSION', '1.0.0' );
+define( 'PAGED_VERSION', '1.0.1' );
 
 /**
- * Admin actions
+ * Register Paged Preview meta box.
  */
+function wppaged_register_meta_boxes() {
+    add_meta_box( 'meta-box-paged', __( 'Paged Preview', 'textdomain' ), 'wppaged_my_display_callback', 'post', 'side', 'high',
```
Comment (jonathanbossenger) [inline paged-wp.php:27]: Can we rename the callback to something more verbose like wppaged_paged_meta_box_display

File: paged-wp.php [hunk 4]
```diff
@@ -18,16 +18,27 @@
...
 /**
- * Adds Paged Preview Button to Page Public Post Box
+ * Meta box display callback.
+ *
+ * @param WP_Post $post Current post object.
  */
-add_action( 'post_submitbox_minor_actions', 'paged_add_paged_preview_button', 10, 1 );
-function paged_add_paged_preview_button( $post ) {
+function wppaged_my_display_callback( $post ) {
+    // Display code/markup goes here. Don't forget to include nonces!
```
Comment (jonathanbossenger) [inline paged-wp.php:41]: Please ensure to add nonce checks for security

File: paged-wp.php [hunk 5]
```diff
@@ -39,11 +50,21 @@ function paged_add_paged_preview_button( $post ) {
...
 	<?php
 }
+ 
+/**
+ * Save meta box content.
+ *
+ * @param int $post_id Post ID
+ */
+function wppaged_save_meta_box( $post_id ) {
+    // Save logic goes here. Don't forget to include nonce checks!
```
Comment (jonathanbossenger) [inline paged-wp.php:65]: Please include the save post logic, as well as nonce checking for security.

**Review by jonathanbossenger — CHANGES_REQUESTED** [review]
Finally, besides moving the main.css from being an external resource to being bundled with the plugin, there seem to be a lot of other asset files that are not necessary for the plugin code to function. Could you please omit them from the PR, thanks

**Comment by jonathanbossenger** [comment]
Hi @typometre, after discussion with @arthurattwell I've review the PR and requested some changes. Please let us know if you can complete these changes so that we can merge in this PR. Thanks

File: paged-wp.php (hunk 5 above)
Comment (jonathanbossenger) [reply paged-wp.php:65]: FYI if there is no intention of actually doing anythnig on the save post, we can leave out this function call

**Comment by jonathanbossenger** [comment]
Hi @typometre I appreciate that open source time is volunteer time, so I just wanted to check in to see if you've seen the review I posted? Do you think you are in a position to be able to make the changes?

**Comment by typometre** [comment]
I was busy and see this just now that I try to read my unread emails. Ok will have a look this week-end.

**Comment by jonathanbossenger** [comment]
No problem, thanks for the update.

**Pull request merged**